Makefile

```diff
@@ -41,6 +41,11 @@ fix-toml: install-taplo-cli
 check-toml: install-taplo-cli
 	taplo check
 
+check-msrv:
```

Comment:
Should we call this from CI as well? Otherwise we're duplicating logic, and it is nice to have the same logic both for local development as CI.

https://github.com/apache/iceberg-rust/blob/bad8e4e6680589082121903871ac67f84abaccc0/.github/workflows/ci.yml#L173C40-L173C61